tests/rptest/tests/pandaproxy_test.py

```diff
@@ -947,6 +949,209 @@ def test_consumer_group_json_v2(self):
         assert rc_res.status_code == requests.codes.no_content
 
 
+class PandaProxyInvalidInputsTest(PandaProxyEndpoints):
+    """
+    Base class for testing how pandaproxy handles invalid messages
+    """
+    def __init__(self, context, **kwargs):
+        super(PandaProxyInvalidInputsTest, self).__init__(context, **kwargs)
+
+    @cluster(num_nodes=3)
+    def test_invalid_member_id(self):
+        """
+        Validates that an invalid member name is rejected
+        """
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+
+        self.logger.debug(
+            "Attempting to create consumer with a new-line character in name")
+        res = self._create_named_consumer(group_id, "my\nconsumer")
+        assert res.status_code == requests.codes.bad_request
+        assert res.json(
+        )["message"] == b'Parameter contained invalid control characters: my\xe2\x90\x8aconsumer'.decode(
+            'utf-8')
+
+    @cluster(num_nodes=3)
+    def test_invalid_group_name(self):
+        """
+        Validates that an invalid consumer group name is rejected
+        """
+        group_id = "My\rconsumer"
+
+        res = self._create_named_consumer(urllib.parse.quote(group_id), "test")
+        assert res.status_code == requests.codes.bad_request
+        assert res.json(
+        )["message"] == f'Invalid parameter \'group_name\' got \'{group_id.translate(CONTROL_CHARS_MAP)}\''
+
+    @cluster(num_nodes=3)
+    def test_bad_arguments_delete_consumer(self):
+        group_name = "My\x02group"
+        sc_res = requests.delete(
+            f"{self._base_uri()}/consumers/{urllib.parse.quote(group_name)}/instances/a",
+            headers=HTTP_REMOVE_CONSUMER_HEADERS)
```

Comment:
nit: We have wrappers in `pandaproxy_test.py` for issuing REST requests. Consider modifying those wrappers instead of directly calling the `requests` lib.
This comment can be addressed in a followup PR though because this is a stylistic change and it should **not** block this PR.